kompari-html/src/report.rs

```diff
@@ -61,16 +61,41 @@ pub fn html_size(width: u32, height: u32, size_limit: u32) -> (Option<u32>, Opti
     }
 }
 
-fn render_difference_image(difference: &Result<ImageDifference, LeftRightError>) -> Markup {
+fn render_difference_image(
+    id: usize,
+    difference: &Result<ImageDifference, LeftRightError>,
+) -> Markup {
     match difference {
-        Ok(ImageDifference::Content { diff_image, .. }) => {
-            let (w, h) = html_size(diff_image.width(), diff_image.height(), IMAGE_SIZE_LIMIT);
-            let mut data = Vec::new();
-            diff_image
-                .write_to(&mut Cursor::new(&mut data), image::ImageFormat::Png)
+        Ok(ImageDifference::Content {
+            rg_diff_image,
+            overlay_diff_image,
+            ..
+        }) => {
+            let (w, h) = html_size(
+                rg_diff_image.width(),
+                rg_diff_image.height(),
+                IMAGE_SIZE_LIMIT,
+            );
+            let mut rg_data = Vec::new();
+            rg_diff_image
+                .write_to(&mut Cursor::new(&mut rg_data), image::ImageFormat::Png)
                 .unwrap();
+            let mut overlay_data = Vec::new();
+            overlay_diff_image
+                .write_to(&mut Cursor::new(&mut overlay_data), image::ImageFormat::Png)
+                .unwrap();
+
             html! {
-                img class="zoom" src=(embed_png_url(&data)) width=[w] height=[h] onclick="openImageDialog(this)";
+                img id=(format!("img-diff1-{}", id)) class="zoom" src=(embed_png_url(&rg_data)) width=[w] height=[h] onclick="openImageDialog(this)";
+                img id=(format!("img-diff2-{}", id)) style="display: none" class="zoom" src=(embed_png_url(&overlay_data)) width=[w] height=[h] onclick="openImageDialog(this)";
```

Comment:
This is partly why I want a Xilem or similar app for Kompari; so that these diffs could be computed on-the-fly. This change effectively doubles the cost of creating a report, right?